### DISCUSSION TITLE
some physical mouse/keyboard problems for Android-> RustDesk

### DISCUSSION THREAD

**Discussion by pokon548** [rustdesk/rustdesk#864] [General]
When I externally connected a set of physical keyboard and mouse to the Android tablet, and tried to manipulate the remote Windows side on the Android side, I found several problems:

The Shift key doesn't seem to "release" when pressed. As long as it is pressed once, the subsequent text received by the Windows side will have the Shift key (for example, when entering 1 on the Android physical keyboard, the Windows side will receive "!"; when entering a, the Windows side will receive "A" ");
The mouse wheel cannot scroll the screen on the Windows side. That is, when I connect the physical mouse on the Android side and scroll the wheel of the physical mouse in the remote control interface of RustDesk, the remote Windows does not respond.
Anyway, thanks for the reply ;)

我在 Android 平板上外接了一套实体键鼠，并尝试在 Android 端操纵远程的 Windows 端时，发现几个问题：

- Shift 键在按下后似乎不会“松开”。只要按下一次，Windows 端后续收到的文本都带有 Shift 按键（例如，在 Android 测实体键盘输入 1 时，Windows 端会收到“！”；输入 a 时，Windows 端将会收到“A”）；
- 鼠标滚轮无法滚动 Windows 侧的画面。即我在 Android 侧接入实体鼠标，并在 RustDesk 的远控界面里滚动实体鼠标的滚轮时，远端 Windows 无反应。

无论如何，感谢回复 ;)

**Comment by rustdesk** [comment]
@Heap-Hop

**Reply by gaetanolip** [reply]
The mouse wheel cannot scroll the screen on the Windows side. same for me, using android app on  xiaomi mi9 t with physical mouse

**Comment by Luqijian** [comment]
Same on linux client while manipulating the remote machine. 
## local machine
OS: Deepin 20.09
RustDesk Version: 1.2.7
## remote machine
OS: Ubuntu 24.04
RustDesk Version: 1.2.7

**Comment by yywxdgy** [comment]
The problem remains in 1.4.0.
## local machine
---
OS: Harmony OS
RustDesk Version: 1.3.8

### remote machine
---
OS: Windows 11
RustDesk Version: 1.4.0
